src/main/scala/org/graphframes/lib/ConnectedComponents.scala

```diff
@@ -327,22 +327,22 @@ object ConnectedComponents extends Logging {
 
         // checkpointing
         if (shouldCheckpoint && (iteration % checkpointInterval == 0)) {
-          // TODO: remove this after DataFrame.checkpoint is implemented
-          val out = s"${checkpointDir.get}/$iteration"
-          ee.write.parquet(out)
-          // may hit S3 eventually consistent issue
-          ee = spark.read.parquet(out)
-
-          // remove previous checkpoint
+          // enable checkpointing if not yet done
+          if (spark.sparkContext.getCheckpointDir.isEmpty) {
```

Comment:
What do you think about moving it outside of the loop?

Reply:
Yeah, if we don't want to control the checkpoint file management? Like we are okay to not clean it up within each iteration? Then we can move it out.

Reply:
I mean only move outside of the loop setting of the spark checkpoint dir (if it is empty). Why should it affect cleaning? Am I missing something?

Reply:
Ok, I see! Thanks for the explanation. The only thing I would like to ask to add is a step when we return checkpointDir to it's initial state (or empty if it was).